langchain/src/chains/sql_db/sql_db_chain.ts

```diff
@@ -206,3 +218,95 @@ export class SqlDatabaseChain extends BaseChain {
     }
   }
 }
+
+interface CreateSqlQueryChainFields {
+  llm: BaseLanguageModel;
+  db: SqlDatabase;
+  prompt?: BasePromptTemplate;
+  /**
+   * @default 5
+   */
+  k?: number;
+  dialect: SqlDialect;
+}
+
+type SqlInput = {
+  question: string;
+};
+
+type SqlInoutWithTables = SqlInput & {
+  tableNamesToUse: string[];
+};
+
+const strip = (text: string) => {
+  // Replace escaped quotes with actual quotes
+  let newText = text.replace(/\\"/g, '"').trim();
+  // Remove wrapping quotes if the entire string is wrapped in quotes
+  if (newText.startsWith('"') && newText.endsWith('"')) {
+    newText = newText.substring(1, newText.length - 1);
+  }
+  return newText;
+};
+
+function difference(setA: Set<string>, setB: Set<string>) {
+  return new Set([...setA].filter((x) => !setB.has(x)));
+}
+
+export async function createSqlQueryChain({
+  llm,
+  db,
+  prompt,
+  k = 5,
+  dialect,
+}: CreateSqlQueryChainFields) {
+  let promptToUse: BasePromptTemplate;
+  if (prompt) {
+    promptToUse = prompt;
+  } else if (SQL_PROMPTS_MAP[dialect]) {
+    promptToUse = SQL_PROMPTS_MAP[dialect];
+  } else {
+    promptToUse = DEFAULT_SQL_DATABASE_PROMPT;
+  }
+
+  if (
+    difference(
+      new Set(["input", "top_k", "table_info"]),
+      new Set(promptToUse.inputVariables)
+    ).size > 0
+  ) {
+    throw new Error(
+      `Prompt must have input variables: 'input', 'top_k', 'table_info'. Received prompt with input variables: ` +
+        `${promptToUse.inputVariables}. Full prompt:\n\n${promptToUse}`
+    );
+  }
+  if (promptToUse.inputVariables.includes("dialect")) {
+    promptToUse = await promptToUse.partial({ dialect });
+  }
+
+  promptToUse = await promptToUse.partial({ top_k: k.toString() });
+
+  const inputs = {
+    input: (x: Record<string, unknown>) => {
+      if ("question" in x) {
+        return `${(x as SqlInput).question}\nSQLQuery: `;
+      }
+      throw new Error("Input must include a question property.");
+    },
+    table_info: async (x: Record<string, unknown>) =>
+      db.getTableInfo((x as SqlInoutWithTables).tableNamesToUse),
+  };
+
+  return RunnableSequence.from([
+    RunnablePassthrough.assign(inputs),
+    (x) => {
+      const newInputs = { ...x };
+      delete newInputs.question;
+      delete newInputs.tableNamesToUse;
+      return newInputs;
+    },
+    promptToUse,
+    llm.bind({ stop: ["\nSQLResult:"] }),
+    new StringOutputParser(),
+    strip,
```

Comment:
This will ruin streaming - do we need to strip things?

Reply:
I tried wrapping it in a `RunnableLambda` and I don't see a difference in traces. Here are two traces, the first is without a `RunnableLambda` and the second is with

1st https://smith.langchain.com/public/b70cf1e3-cf53-4e3e-9667-8f22e034d968/r
2nd https://smith.langchain.com/public/0ae75057-c01f-43d5-9b63-6f1761fc0da6/r

The `strip` call is the last in both